### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by DumouchelP**

### CODE REVIEW

File: tutorials/tutorial10.py [hunk 1]
```diff
@@ -228,7 +229,7 @@ def plot_data(self):
                 edges.append(edge)
         return [plot_data.graph.NetworkxGraph(gearbox_graph)]
 
-    def to_dict(self, subobjects_id={}):
+    def to_dict(self, use_pointers: bool = True, memo=None, path: str = '#'):
```
Comment (AxelRinghausen) [inline tutorials/tutorial10.py:232]: All to_dict and dict_to_object methods should be deleted.

File: tutorials/tutorial1_rivetedassembly.py [hunk 2]
```diff
@@ -315,25 +333,27 @@ def _fatigue_resistance(self):
...
+        thickness = vm.O3D + 2*pan_vm[0].extrusion_vector
+        # for panel, pan_vm, hole in zip(self.panel_combination.panels, self.panel_combination.volmdlr_primitives(),
+        #                                holes):
+        #     center, dir1, dir2 = pan_vm.plane_origin, pan_vm.x, pan_vm.y
+        #     contour, dir3 = panel.contour(), pan_vm.extrusion_vector
+        #     thickness += dir3
+        #
+        #     pan_hole = p3d.ExtrudedProfile(center, dir1, dir2, contour, [], dir3,
+        #                                    name='extrusion')
+        #     primitives.append(pan_hole)
```
Comment (AxelRinghausen) [inline tutorials/tutorial1_rivetedassembly.py:350]: Clean up

File: tutorials/tutorial1_rivetedassembly.py [hunk 3]
```diff
@@ -369,12 +389,15 @@ def define_grid(self, contour: vm.wires.Contour2D, number_rivet1: int, number_ri
                 grids.append(vm.Point2D(xmin + ratio1 * (n1 + 1), ymin + ratio2 * (n2 + 1)))
         return grids
 
-    def generate(self) -> List[PanelAssembly]:
+    def generate(self, progress_callback=lambda x: 0) -> List[PanelAssembly]:
```
Comment (AxelRinghausen) [inline tutorials/tutorial1_rivetedassembly.py:392]: progress_callback input might interfere with usage in platform ?

**Review by AxelRinghausen — CHANGES_REQUESTED** [review]

File: tutorials/tutorial1_rivetedassembly.py (hunk 3 above)
Comment (younesdessia) [reply tutorials/tutorial1_rivetedassembly.py:392]: I have the same in other __init__ and it works in the platform, whith the workflow of this exemple too, I can remove it if you want !

**Pull request merged**